core/rio/turtle/src/main/java/org/eclipse/rdf4j/rio/turtle/TurtleWriter.java

```diff
@@ -701,4 +735,109 @@ private void wrapLine(boolean space) throws IOException {
 			writer.write(" ");
 		}
 	}
+
+	/*
+	 * not synchronized, assumes calling method has obtained a lock on bufferLock
+	 */
+	private void processBuffer() throws RDFHandlerException {
+		if (!isBuffering()) {
+			return;
+		}
+
+		if (this.getRDFFormat().supportsContexts()) { // to allow use in Turtle extensions such a TriG
+			// primary grouping per context.
+			for (Resource context : bufferedStatements.contexts()) {
+				Model contextData = bufferedStatements.filter(null, null, null, context);
+				Set<Resource> processedSubjects = new HashSet<>();
+				Optional<Resource> nextSubject = nextSubject(contextData, processedSubjects);
+				while (nextSubject.isPresent()) {
+					processSubject(contextData, nextSubject.get(), processedSubjects);
+					nextSubject = nextSubject(contextData, processedSubjects);
+				}
+			}
+		} else {
```

Comment:
Does this now support pretty printing TRIG? That would be awesome.

Reply:
I didn't check - I'm pretty sure TriGWriter already supported pretty printing to some extent though.